ppexpect.py

```diff
@@ -0,0 +1,90 @@
+# SPDX-License-Identifier: GPL-2.0-only
+
+# Poor Python Expect.
+# Simplified expect to run a process
+
+import errno
+import fcntl
+import os
+import select
+import subprocess
+import sys
+import time
+
+__all__ = ['Process', 'TimeoutError']
+
+def set_nonblocking(fd):
+    fl = fcntl.fcntl(fd, fcntl.F_GETFL)
+    fcntl.fcntl(fd, fcntl.F_SETFL, fl | os.O_NONBLOCK)
+
+if sys.version_info[0] < 3:
+    class TimeoutError(OSError):
+        def __init__(self):
+            super(TimeoutError, self).__init__(errno.ETIMEDOUT, None)
+else:
+    class TimeoutError(TimeoutError):
```

Comment:
Why is this redefinition of TimeoutError needed?

Reply:
It's for Python 2/3 compatibility.

Reply:
Yeah I can see that. I guess I was wondering why TimeoutError needs to be redefined in the Py3 case when it already exists.

Reply:
Now I got the question. To be exported and used by the test that's not attached at the moment! I'll update to a committable state